### DISCUSSION TITLE
Work download-counter

### DISCUSSION THREAD

**Discussion by deamon2k** [rejetto/hfs#240] [Ideas]
Is it possible that the counter would increase not only when the file is downloaded, but also when play?

**Comment by rejetto** [comment]
I expect it to count playings as well, because HFS makes no difference.
Playing is just the browser downloading not to file but "to video", so the difference is on the browser side, not on server side.
Are you experiencing differently?

**Reply by deamon2k** [reply]
The counter works weird.

folder name in settings
trailers

"/trailers/Elden%20Ring%20DLSS%203%20Mod%204K%20Test%20%5BMax%20Settings,%20RTX%204090%5D.mp4": 39
"/Trailers/Elden%20Ring%20DLSS%203%20Mod%204K%20Test%20%5BMax%20Settings,%20RTX%204090%5D.mp4": 20

Some lines are in quotes and some are not.

/Trailers/TATTOO%20PARLOR%20TRICK!%20MAGICIAN%20BECOMES%20MARKED%20MAN!.mp4: 8
/trailers/TATTOO%20PARLOR%20TRICK!%20MAGICIAN%20BECOMES%20MARKED%20MAN!.mp4: 7

The file has all the counters for the files, but very few are displayed in the sheet. There are no visible reasons.

**Comment by rejetto** [comment]
the quotes in the file are added automatically (not by me) to comply to "yaml" file format.
What I see is that names are duplicated because of the lowercase/uppercase T in "trailers". This is clearly bad.
Are you on Windows ?
And what "player" plugin are you using?

**Reply by deamon2k** [reply]
Windows
integrated player

**Reply by rejetto** [reply]
by "integrated player" you mean that you click the file name and then "open", right?

**Reply by deamon2k** [reply]
Right

**Reply by deamon2k** [reply]
I not use player in plugins

**Comment by rejetto** [comment]
is the actual name of the folder Trailers with capital T ?

**Reply by deamon2k** [reply]
_Yes_
Sory!!!
**No**
name directory - trailers

**Reply by deamon2k** [reply]
![image](https://user-images.githubusercontent.com/86757569/236671556-d89c821c-106e-4b8b-8154-81471a40cb08.png)

**Reply by rejetto** [reply]
I wonder where the capital T comes from... 
is it possible that you are typing the capital T yourself? I mean, with your keyboard?

**Reply by rejetto** [reply]
or anyway, let me know if you happen to see/understand how the capital T comes to be

**Reply by deamon2k** [reply]
No, I not typing.
This is part counters.yaml

/trailers/%D0%9F%D0%9E%D0%92%D0%A2%D0%9E%D0%A0%D0%98%D0%9B%20%D0%A1%D0%90%D0%9C%D0%AB%D0%95%20%D0%A0%D0%95%D0%94%D0%9A%D0%98%D0%95%20%D0%91%D0%A3%D0%A0%D0%93%D0%95%D0%A0%D0%AB%20%D0%92%20%D0%9C%D0%98%D0%A0%D0%95%20%D0%98%D0%97%20McDonald%E2%80%99s%20%20Burger%20King%20%20KFC%202.mp4: 2
"/trailers/%D0%92%D0%B0%D0%B9%D0%BA%D1%83%D0%BB%D0%B5%20%D0%BE%20%D1%82%D0%BE%D0%BC,%20%D0%BA%D0%B0%D0%BA%20%D0%B7%D0%B0%D0%BA%D0%BE%D0%BD%D1%87%D0%B8%D1%82%D1%8C%20%D0%B2%D0%BE%D0%B9%D0%BD%D1%83%20%23shorts.mp4": 12
"/trailers/%D0%91%D0%B5%D0%B6%D0%B5%D0%BD%D0%BA%D0%B0%20%D0%B8%D0%B7%20%D0%9C%D0%B0%D1%80%D0%B8%D1%83%D0%BF%D0%BE%D0%BB%D1%8F%20%D0%BE%20%D1%82%D0%BE%D0%BC,%20%D1%87%D0%B5%D0%B3%D0%BE%20%D0%B6%D0%B5%D0%BB%D0%B0%D0%B5%D1%82%20%D1%80%D0%BE%D1%81%D1%81%D0%B8%D1%8F%D0%BD%D0%B0%D0%BC.mp4": 33
"/trailers/Apple%20%D0%BD%D0%B8%D0%BA%D1%82%D0%BE%20%D0%BD%D0%B5%20%D0%BF%D0%BE%D0%BA%D1%83%D0%BF%D0%B0%D0%B5%D1%82,%20%D1%81%D0%B5%D1%80%D0%B8%D0%B0%D0%BB%20%D0%BF%D0%BE%20%D0%93%D0%B0%D1%80%D1%80%D0%B8%20%D0%9F%D0%BE%D1%82%D1%82%D0%B5%D1%80%D1%83,%20%D0%B1%D0%BB%D0%BE%D0%BA%D0%B8%D1%80%D0%BE%D0%B2%D0%BA%D0%B0%20ChatGPT%20%D0%B8%20%D0%BD%D0%B5%D0%B9%D1%80%D0%BE%D1%81%D0%B5%D1%82%D1%8C%20%D0%BE%D1%82%20%D0%AF%D0%BD%D0%B4%D0%B5%D0%BA%D1%81%D0%B0.mp4": 10
/trailers/SAKRA%20Official%20Trailer%20(2023).mp4: 10
/trailers/Top%20Gun%20Maverick%20(2022)%20%20Special%20Announcement%20From%20the%20Set%20of%20%23MissionImpossible%20(Tom%20Cruise).mp4: 6
/trailers/%D0%A1%D0%B0%D0%BC%D0%B0%D1%8F%20%D0%B4%D0%BE%D1%80%D0%BE%D0%B3%D0%B0%D1%8F%20%D0%B8%20%D1%81%D0%B0%D0%BC%D0%B0%D1%8F%20%D0%B4%D0%B5%D1%88%D0%B5%D0%B2%D0%B0%D1%8F%20%D1%83%D0%BB%D0%B8%D1%87%D0%BD%D0%B0%D1%8F%20%D0%B5%D0%B4%D0%B0%20%D0%91%D0%B0%D0%BD%D0%B3%D0%BA%D0%BE%D0%BA%D0%B0%20@staspognali.mp4: 46
/Trailers/%D0%A3%D1%85%D0%BE%D0%B4%D0%B8%D1%82%D0%B8.mp4: 3
/trailers/Can%20AI%20code%20Flappy%20Bird%20Watch%20ChatGPT%20try.mp4: 13
/trailers/%D0%9A%D0%B0%D0%BA%D0%BE%D0%B9%20%D0%BA%D0%B0%D0%BB%D0%B8%D0%B1%D1%80%20%D0%B7%D0%B0%D1%85%D0%BE%D0%B4%D0%B8%D1%82%20%D0%B3%D0%BB%D1%83%D0%B1%D0%B6%D0%B5%20%D0%B2%D1%81%D0%B5%D1%85%20%20%D0%A0%D0%B0%D0%B7%D1%80%D1%83%D1%88%D0%B8%D1%82%D0%B5%D0%BB%D1%8C%D0%BD%D0%BE%D0%B5%20%D1%80%D0%B0%D0%BD%D1%87%D0%BE.mp4: 64
/Trailers/La%20diferencia%20entre%20la%20velocidad%20de%20la%20luz%20y%20el%20sonido.%20%20-.mp4: 2
/trailers/renfield_zajadromcagesfx_1920.mp4: 27
/Trailers/IMG_5084.MP4: 5
/Trailers/IMG_8167.MP4: 6
/Trailers/%D0%92%20%D0%A1%D0%A8%D0%90%20%D1%81%D0%B1%D0%B5%D0%B6%D0%B0%D0%B2%D1%88%D0%B8%D1%85%20%D0%B7%D0%B0%D0%BA%D0%BB%D1%8E%D1%87%D0%B5%D0%BD%D0%BD%D1%8B%D1%85%20%D0%BF%D0%BE%D0%B9%D0%BC%D0%B0%D0%BB%D0%B8%20%D0%B2%20%D0%B1%D0%BB%D0%B8%D0%BD%D0%BD%D0%BE%D0%BC%20%D1%80%D0%B5%D1%81%D1%82%D0%BE%D1%80%D0%B0%D0%BD%D0%B5%20%23shorts.mp4: 5
/Trailers/%D0%A2%D0%B2%D0%BE%D1%8F%20%D0%BB%D0%B8%20%D0%B2%20%D1%8D%D1%82%D0%BE%D0%BC%20%D0%B2%D0%B8%D0%BD%D0%B0%20%23shorts.mp4: 4
/Trailers/%D0%A1%D0%B5%D0%BA%D1%80%D0%B5%D1%82%20%D0%BF%D0%BE%D1%85%D1%83%D0%B4%D0%B5%D0%BD%D0%B8%D1%8F%20%D0%A1%D0%BE%D0%B1%D1%87%D0%B0%D0%BA.mp4: 4
/Trailers/%D0%9D%D0%B5%D0%BA%D0%BE%D0%B3%D0%BB%D0%B0%D0%B9%20%D0%97%D0%B0%20%D0%B4%D0%B2%D0%B0%20%D0%BC%D0%B5%D1%81%D1%8F%D1%86%D0%B0%20%D1%8F%20%D0%B7%D0%B0%D1%80%D0%B0%D0%B1%D0%BE%D1%82%D0%B0%D0%BB%20%D0%BF%D0%BE%D0%BB%D1%82%D0%BE%D1%80%D0%B0%20%D0%BC%D0%B8%D0%BB%D0%BB%D0%B8%D0%BE%D0%BD%D0%BE%D0%B2%20%D0%B4%D0%BE%D0%BB%D0%BB%D0%B0%D1%80%D0%BE%D0%B2%20%23shorts.mp4: 5
"/Trailers/%D0%9A%D0%BB%D0%B8%D0%BC%20%D0%96%D1%83%D0%BA%D0%BE%D0%B2.%20%D0%9C%D0%B8%D1%84%D1%8B%20%D0%BE%20%D0%BD%D0%BE%D0%B6%D0%B0%D1%85%20%D0%B3%D0%B4%D0%B5%20%D0%BF%D1%80%D0%B0%D0%B2%D0%B4%D0%B0,%20%D0%B0%20%D0%B3%D0%B4%D0%B5%20%D0%BB%D0%BE%D0%B6%D1%8C%201%20%D1%87%D0%B0%D1%81%D1%82%D1%8C!.mp4": 12
"/Trailers/%D0%9B%D1%8E%D0%B4%D0%B8,%20%D0%9A%D0%BE%D1%82%D0%BE%D1%80%D1%8B%D0%B5%20%D0%92%D1%8B%D1%88%D0%BB%D0%B8%20%D0%BD%D0%B0%20%D0%9D%D0%BE%D0%B2%D1%8B%D0%B9%20%D0%A3%D1%80%D0%BE%D0%B2%D0%B5%D0%BD%D1%8C.%20%D0%A1%D0%B0%D0%BC%D1%8B%D0%B5%20%D0%A3%D0%B4%D0%B8%D0%B2%D0%B8%D1%82%D0%B5%D0%BB%D1%8C%D0%BD%D1%8B%D0%B5%20%D0%9D%D0%B0%D0%B2%D1%8B%D0%BA%D0%B8%20%D0%B8%20%D0%A2%D0%B0%D0%BB%D0%B0%D0%BD%D1%82%D1%8B.mp4": 9
"/trailers/%D0%9D%D0%B0%D1%81%D1%82%D1%83%D0%BF%D0%BB%D0%B5%D0%BD%D0%B8%D0%B5%20%D0%B8%20%D0%BD%D0%B0%D0%BA%D0%B0%D0%B7%D0%B0%D0%BD%D0%B8%D0%B5,%20%D0%9F%D0%B0%D1%88%D0%B8%D0%BD%D1%8F%D0%BD%20%D0%B8%20%D0%9D%D0%90%D0%A2%D0%9E,%20%D0%92%D0%B7%D1%8F%D0%BB%D0%B8%D1%81%D1%8C%20%D0%B7%D0%B0%20%D0%9C%D0%B5%D0%BB%D0%B0%D0%B4%D0%B7%D0%B5%20%20%D0%93%D0%B0%D0%BB%D0%BE%D0%BF%D0%BE%D0%BC%20%D0%BF%D0%BE%20%D0%95%D0%B2%D1%80%D0%BE%D0%BF%D0%B0%D0%BC%20%23926.mp4": 17
/Trailers/%D0%9C%D0%B8%D0%BB%D1%8C%20%D0%92-12%20-%20%D1%81%D0%BE%D0%B2%D0%B5%D1%82%D1%81%D0%BA%D0%B8%D0%B9%20%D0%B2%D0%B8%D0%BD%D1%82%D0%BE%D0%BA%D1%80%D1%8B%D0%BB%D1%8B%D0%B9%20%D1%82%D0%B8%D1%82%D0%B0%D0%BD.mp4: 16
"/Trailers/%D0%91%D0%B5%D0%B6%D0%B5%D0%BD%D0%BA%D0%B0%20%D0%B8%D0%B7%20%D0%9C%D0%B0%D1%80%D0%B8%D1%83%D0%BF%D0%BE%D0%BB%D1%8F%20%D0%BE%20%D1%82%D0%BE%D0%BC,%20%D1%87%D0%B5%D0%B3%D0%BE%20%D0%B6%D0%B5%D0%BB%D0%B0%D0%B5%D1%82%20%D1%80%D0%BE%D1%81%D1%81%D0%B8%D1%8F%D0%BD%D0%B0%D0%BC.mp4": 17
/Trailers/%D0%9A%D0%BE%D1%80%D0%BE%D0%BB%D0%B8%20%D0%BF%D0%BB%D0%B0%D0%B3%D0%B8%D0%B0%D1%82%D0%B0%20-%20Morgenshtern!.mp4: 13
/Trailers/%D0%A7%D1%82%D0%BE%20%D1%81%D0%BB%D1%83%D1%87%D0%B8%D0%BB%D0%BE%D1%81%D1%8C%20%D1%81%20%D0%9C%D0%B0%D0%BAS%D0%B8%D0%BC%20%23%D0%BC%D0%B0%D0%BA%D1%81%D0%B8%D0%BC%20%23%D1%81%D0%BE%D0%B1%D1%87%D0%B0%D0%BA.mp4: 14
/Trailers/%D0%9F%D0%B5%D1%81%D0%BD%D1%8F%20Shallow%20%D0%BD%D0%B0%20%D1%80%D1%83%D1%81%D1%81%D0%BA%D0%BE%D0%BC!%F0%9F%91%8D.mp4: 11
/Trailers/%D0%A2%D0%B5%D1%81%D1%82%D0%B8%D1%80%D1%83%D0%B5%D0%BC%20%D0%9F%D0%BE%D0%BB%D0%B8%D1%86%D0%B5%D0%B9%D1%81%D0%BA%D0%B8%D0%B9%20%D0%A9%D0%B8%D1%82.mp4: 1
/upload/%C2%AB%D0%9B%D0%B5%D1%81%D0%BD%D0%BE%D0%B9%20%D0%BE%D0%BB%D0%B5%D0%BD%D1%8C%C2%BB%20(%D0%9A_%D1%84%20%D0%9E%D1%85%20%D1%83%D0%B6%20%D1%8D%D1%82%D0%B0%20%D0%9D%D0%B0%D1%81%D1%82%D1%8F!)%20-%20%D0%94%D0%B0%D1%80%D0%B8%D1%88%D0%B0%20%D0%9A%D0%B8%D0%BD%D1%8C%D0%BE%D0%BD%D0%B5%D1%81%20%D1%84%D1%80%D0%B0%D0%B3%D0%BC%D0%B5%D0%BD%D1%82%20%D0%BF%D1%80%D1%8F%D0%BC%D0%BE%D0%B3%D0%BE%20%D1%8D%D1%84%D0%B8%D1%80%D0%B0.mp4: 40

**Reply by deamon2k** [reply]
I tried to analyze. And download and open, and then look in counters.yaml. It is not yet clear why there are capital letters or quotation marks.

**Comment by rejetto** [comment]
this version should fix the problem
https://github.com/rejetto/hfs/discussions/241

**Reply by Dane501** [reply]
I cannot find a download counter in 0.52.9.
Nor do i see a plugin for keeping track of how many times each file has been downloaded.
Is there one available?

Great job with HFS, by the way.
I had used the old v2 for many years.

:^)

**Comment by rejetto** [comment]
hey @Dane501 
download-counter plugin comes with HFS 3 when you install it.
so you'll find it in admin-panel > plugins > installed

**Reply by Dane501** [reply]
Thank you for replying.
Hmmm... The only plug-in listed there is "mark-news", which i installed.
Where are the download counts displayed?
If i can see them there, then i will know they are being counted.

:^)

**Reply by rejetto** [reply]
something is wrong.
how did you install hfs?

**Comment by Dane501** [comment]
I installed from the 0.52.9 hfs-windows.zip file i got at https://github.com/rejetto/hfs/releases.

**Reply by Dane501** [reply]
If it helps, you can see my server running here:  [da.getmyip.com](http://da.getmyip.com/)

**Comment by Dane501** [comment]
Should just uninstall it, and start fresh?

**Comment by rejetto** [comment]
inside the zip you have hfs.exe and "plugins" folder.
maybe you skipped or deleted the folder.

**Reply by rejetto** [reply]
you can just copy the content of the folder from the zip, even without restarting hfs

**Comment by Dane501** [comment]
That was it!
I had moved the folders and that one didn't come with.
Thanks a bunch.

:^)

**Comment by Dane501** [comment]
Is there a way to have the download count seen along with the file name, size, and date?

**Comment by rejetto** [comment]
yes, just configure the plugin
<img width="346" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/b409f334-bcdf-4682-bac2-a1bfec731e60">

**Reply by rejetto** [reply]
"where" could be clearer

**Comment by Dane501** [comment]
I mean, not in the pop-up window, but in the list of file names.

**Reply by rejetto** [reply]
the list indeed.
did you configure it?

**Comment by Dane501** [comment]
You're the best!
Thanks.
I'll stop bothering you now.

:^)

**Comment by rejetto** [comment]
no prob

i'll change it like this
<img width="199" alt="image" src="https://github.com/rejetto/hfs/assets/1367199/7ace8d00-08ca-49e9-94ae-3550e8b30b48">

**Comment by Dane501** [comment]
I was just used to the old v2 HFS.
I used it for many years with millions of downloads.
But someone hacked it last week and deleted all my files, which prompted me to look into the new version.

:^)

**Reply by rejetto** [reply]
always stay updated when you have a server online!
it's a jungle out there

**Reply by Dane501** [reply]
Is the Github sponsor option the best/only way to make a one time donation?

**Comment by Dane501** [comment]
Is the Github sponsor option the best/only way to make a one time donation?

**Reply by rejetto** [reply]
yes, it's a perfectly fine method, but you can use paypal if you like: a@rejetto.com

**Reply by rejetto** [reply]
i got your big donation! thank you very much ❤️

**Comment by Dane501** [comment]
Worth every penny.

Question: I have zipping turned off, so the button is greyed out.
But "Select" is still an option.
Isn't "select" so you can check which items to zip together?
If so, is there a way to disable it along with Zip?

:^)
Dave

**Reply by rejetto** [reply]
select is also for deleting (if you have permission).
i'm not totally sure what to do about it, but you can immediately hide it with a simple style rule
```css
#select-button { display: none; }
```

**Comment by Dane501** [comment]
Ahh, thanks.

:^)

**Comment by Dane501** [comment]
Here's my "Custom HTML" page:

<style>
#select-button { display: none; }
#zip-button { display: none; }
:root { --bg: #fff; --text: #000; }
a { color: #000 }
</style>


I'm a regular code monkey now.
ha ha

:^)

**Reply by rejetto** [reply]
very well :)

**Comment by Dane501** [comment]
Not calling you a code monkey. (no offense)
I just was thinking of this song:

https://www.youtube.com/watch?v=j4TnhemCEmc

**Comment by charettepa** [comment]
I just tried to upgrade from 0.52.9 to 0.53
the download counter now throws errors
and no longer displays download count

```
plugin download-counter: TypeError: Cannot convert undefined or null to object
    at Function.entries (<anonymous>)
    at exports.init (D:\hfs-windows\plugins\download-counter\plugin.js:25:36)
    at async initPlugin (C:\snapshot\hfs\dist\src\plugins.js:128:30)
    at async start (C:\snapshot\hfs\dist\src\plugins.js:429:13)
```

Reverting to 0.52.9 stops the problem
and thecounter works again

**Reply by rejetto** [reply]
how did you upgrade? manually or using the "check for updates" button?

**Reply by charettepa** [reply]
using the check for updates button
the update was succesful

**Reply by charettepa** [reply]
any further thoughts on the download counter and 0.53

**Reply by rejetto** [reply]
not yet, i studied the thing but couldn't find a way to reproduce, or even understand what happened to you

**Reply by charettepa** [reply]
it happens automatically at start once im on 0.53
if i restore by backup of 0.52.9
it starts normally again

i have tried to upgrade more than once and get the same result everytime

if i am reading the error correctly
it seems to possibly be related to folders/items that allow anonymous(anyone)

it also may not be fully related to 0.53 upgrade and rather the version bump of the counter
which is automatically updated when upgrading from 0.52.9 > 0.53

**Reply by rejetto** [reply]
yes, that's what i tried.
i tried with a clean 0.52.9, i used the downoad-counter plugin, then upgraded to 0.53, and the output was
18:42:12 loading plugin download-counter
18:42:12 loading plugin antibrute
18:42:12 plugin download-counter: data converted

the counters were not reset, and it's working in the 0.53.
I'm quite sure your problem is bound to your file counters.yaml, there must be some problem with it.
You can send it to me for inspection a@rejetto.com

**Reply by charettepa** [reply]
thanks for that
ill check
if it boils down to resetting the counters
ie. replacing the file
i may do that

but ill also see if i can just manually change the file
if i still cant figure it out, ill send you the file

**Reply by rejetto** [reply]
anyway, i expect that error to be only related to the conversion of the counters file.
failing the conversion, i expect the counting to just start over, but to work nonetheless.
the old file is a standard yaml text file, so you may be able to find the problem inspecting it yourself.
googling i've just found this: https://www.yamllint.com/

**Reply by charettepa** [reply]
just looking now and every single item has a value
ie. none are null and none are 0 and all are valid normal numerical values
they are all 1 or higher

i will try to upgrade again soon and see what happens

**Reply by charettepa** [reply]
i tried yamlint
it finds no errors
but if i allow it to reformat and identify aliases
it adds 1 line at the beggining
ie. line 1
with nothing more than ---

**Reply by charettepa** [reply]
Sorry for the very long delay in my reply, we had a few critical issues that took me away from this for a little while.  I do have it working now.  i repeated the update several times, with the same issue every time.  Then I realized what was happening.  every time i would attempt to update, someting was happening where the download counter file was being flushed to an empty file.  i noticed this when the file was down to 0bytes instead of 10kb.  to solve it, i stopped HFS 0.53 replaced the file from the backup i took just a minute before, and then started HFS.  at this point the data converted and it now works. i repeated the process again just to make sure. for some reason on intial startup after an upgrade the file gets wiped/cleared, not sure why, but stopping replacing from backup and starting gets it working.

**Comment by Dane501** [comment]
I was wondering if you could point me to a simple walk-through on how to
set up letsencrypt, or Certbot, or whoever on HFS.
I searched your discussion group but what was there was over my head.

thanks
:^)
Dave

**Reply by rejetto** [reply]
hey! just go on Admin-panel / internet / HTTPS
fill domain, email, enable "automatic", and click request.
you are done.

**Comment by Dane501** [comment]
Thank you for replying!
The request fails even though the settings seem right.
A pop-up says "Error: Configure your domain to point to 70.125.8.199
(currently on 132.226.44.1)
But i click "Verify" and it confirms that i am on 70.128.8.199

Do i have my settings wrong somewhere?

[image: HFS-.jpg]

**Reply by rejetto** [reply]
da.getmyip.com is pointing to 132.226.44.1
da-ip.getmyip.com is pointing to 70.125.8.19
but they are different domains.
it is possible you were mixing things here

**Comment by Dane501** [comment]
Also: My IP Address is dynamic, so i use DYNS and their Dyn Updater tool to
keep my domain name good. (da.getmyip.com)

Could that be the issue?

And i have HFS version 0.52.9

**Comment by rejetto** [comment]
dynamic ip is normally ok, although during changes it may happen that the domain is stale for a bit, because of cache on servers.

> But i click "Verify" and it confirms that i am on 70.128.8.199

it seems incoherent, but you are using an old version, so first move is to try updating

for future requests, start a new discussion, as it's easier to follow; this thread is for a different topic.

**Comment by Dane501** [comment]
HFS Home "Check for updates" says 0.55.9 is available.
It downloads it then after a few seconds it says "Warning. This is taking
too long, please check your server" and the greyed-out Install button just
spins, and i have to reboot.
After the reboot i right-click the "Check for updates" button and provide
the zip link from github for the latest windows release (0.55.4).
HFS downloads that, and again, after just a few seconds it says "Warning.
This is taking too long, please check your server", so i wait and wait
again but have to reboot again.

0.55.4 is now in my HFS folder and i run it, but the HFS server does not
work.
I go back to 0.52.9, and now it does not work either, even though HFS Home
says "Server is working"*.*

I have updated Firefox.
I have rebooted my PC multiple times.
I have even rebooted my router.

I also clicked on the link on the Internet page above where it says "No
certificate configured".
I forget what it said, but now it says "Missing certificate" and i can't
see a way to un-check that button.
On the Options page i did a "Self-Cert" but that didn't fix it the sever,
and now it is on port 443 for some reason

If i put in my ip address in the address bar on the server PC my server
shows up, and i can see my shared files.
But if i put my ip address (or the domain name) in the address bar of a
browser on a different PC it does not work.

I have given it a coupe of hours to maybe settle-in, but no joy.
So now my HFS server no longer works.
I really don't want to delete all the files and start from scratch.

This is why i don't do updates unless absolutely necessary.   :^(

**Comment by rejetto** [comment]
what is your Windows version?
and you say "run it, but does not work", but that's very little to work on. At least you should see what error message you get.

**Comment by Dane501** [comment]
windows 10
I run the hfs.exe and it opens, and says "Server is working", but it can't
be accessed.

**Reply by rejetto** [reply]
we should first try to solve with the latest version, what happens when you run it?

**Comment by Dane501** [comment]
If you want to see for yourself, i have Anydesk and Rustdesk

**Reply by rejetto** [reply]
i'm going to bed now, tomorrow i'll be working, but i can tell you as i find some time. You can write me a@rejetto.com

**Comment by Dane501** [comment]
i am running 0.55.4 (the latest on github)
It opens and the Home page says "Server is working"

**Reply by rejetto** [reply]
ok, so it is running, but you can't access from other computers, right?

**Reply by rejetto** [reply]
try disabling your firewall, and see if it solves. You may have blocked hfs by accident.

**Comment by Dane501** [comment]
Correct.

**Comment by Dane501** [comment]
Go to bed.
We can work on it tomorrow.

:^)

**Comment by Dane501** [comment]
my domain is da.getmyip
But a while back a guy at DYNS added da-ip.getmyip to fix some issue
so now both of them have been going to my server.

**Reply by rejetto** [reply]
i can tell you they point to different IP addresses.
hfs was telling you the truth
<img width="208" alt="image" src="https://github.com/user-attachments/assets/bcd08f80-546a-41ec-95c3-56fca18660dd" />

**Comment by Dane501** [comment]
at DYNS da.getmyip is web-hopped to da-ip.getmyip

**Comment by Dane501** [comment]
so either one gets you here

**Comment by Dane501** [comment]
and da-ip is showing 70.125.8.199, which is what HFS is showing as my ip

**Comment by Dane501** [comment]
and da-ip is showing 70.125.8.199 at DYNS, which is what HFS is showing as
my ip

**Comment by Dane501** [comment]
I turned the firewall off
but still no joy

**Comment by Dane501** [comment]
70.125.8.199 should work even if DYNS is not working, but it doesn't

**Comment by Dane501** [comment]
sleep on it.
Tomorrow is another day.

:^)

**Reply by rejetto** [reply]
just for the record, and the public: 😄
we solved using anydesk.
Problem was wrong domain name, and missing router configuration.
Domains pointing to a "web hop" don't really point to your machine, so HFS cannot generate the certificate for them.
Now working. Cheers

**Comment by Dane501** [comment]
YES, definitely all my fault.

:^)
